lg_common/src/lg_common/uscs_service.py

```diff
@@ -61,37 +61,38 @@ def __init__(self,
 
         self.state = self._grab_state(initial_state_scene_url)
 
-        if not self.on_offline_state:
+        if self.state is None:
+            self.state = USCSMessageResponse()
+
+        if self.on_offline_state is None:
             """
             By default, when offline, go to initial state if there is no offline state
             explicitly specifyed.
             """
             self.on_offline_state = self.state
 
-        if self.state is None:
-            self.state = USCSMessageResponse()
-
-    def handle_connectivity_message(self, message):
+    def handle_offline_message(self, message):
         """
         Accepts online/offline state message
         and emits appropriate message from ivars
 
-        If data == True then emit on_online_state message
-        If data == False then emit on_offline_state message
+        If data == True then emit on_offline_state message
+        If data == False then emit on_online_state message
         """
+        rospy.loginfo("Incoming message: %s. Currently offline: %s" % (message, self.offline))
         if self.director_scene_publisher:
-            if message.data is True and self.on_online_state:
+            self.offline = message.data
```

Comment:
This is going to overwrite the state and lead to a following bug:
- you are online
- you emit offline = False message
- you revert to default state

^ that's bad - you should listen on messages and set the state idempotently. So set the offline/online ivar only when it's different than the state.